src/transformers/models/codegen/modeling_codegen.py

```diff
@@ -303,6 +313,7 @@ class CodeGenPreTrainedModel(PreTrainedModel):
     supports_gradient_checkpointing = True
     _no_split_modules = ["CodeGenBlock"]
     _skip_keys_device_placement = "past_key_values"
+    _supports_cache_class = True
```

Comment:
Don't forget 

```py
_supports_quantized_cache = True
_supports_static_cache = True
```

(if appropriate, on this and other models)

Reply:
Oh, yes, forgot about `_supports_quantized_cache` flag. But I am not sure about the "static_cache" flag. Will it imply that models are fullgraph compilable, cause we didn't test it

in any case I'll make a following PR to check fullgraph compile and if it works as-is add the tests in each modeling

Reply:
👍 

We should add a mixin test for fullgraph compilation when `_supports_static_cache = True`

Reply:
Yes, I will test and add a mixin test (one lightweight and one slow maybe) after this PR is merged. 

> I think _supports_static_cache and _supports_cache_class  will mean the same thing 

Btw, GIT will be an exception which supports cache class but not static cache as it has some special attn mask preparation steps